test/unit/BPool/BPool_Bind.tree

```diff
@@ -0,0 +1,26 @@
+BPool::Bind
+├── when reentrancy lock is set
+│   └── it should revert
+├── when caller is NOT controller
+│   └── it should revert
+└── when caller is controller
+    ├── when token is already bound
+    │   └── it should revert
+    ├── when pool is finalized
+    │   └── it should revert
+    ├── when MAX_BOUND_TOKENS tokens are already bound
+    │   └── it should revert
+    ├── when token weight is too low
+    │   └── it should revert
+    ├── when token weight is too high
+    │   └── it should revert
+    ├── when too little balance is provided
+    │   └── it should revert
+    ├── when weight sum exceeds MAX_TOTAL_WEIGHT
+    │   └── it should revert
+    └── when token can be bound
+        ├── it sets reentrancy lock
+        ├── it increments _totalWeight
+        ├── it calls _pullUnderlying
+        ├── it adds token to the list
```

Comment:
`+ it binds token` ? (and show it does have `bound: true, index: _index, denorm: weight`
"the list" i understand it as `_tokens` array

Reply:
updating the comment/tree since it was the only place where I was referring to this as 'the list' instead of 'the tokens array'

> `+ it binds token`

those assertions in particular are covered as `it sets the token record` since binding the token is this whole process (updating `_totalWeight`, `_tokens`, etc). 
There are assertions for both `bound` and `denorm`, should I add something for `index`? that'd require adding some tokens to the array as a precondition, but would assert we're setting the correct index and not just 0